ambry-store/src/main/java/com/github/ambry/store/BlobStore.java

```diff
@@ -470,23 +474,38 @@ public void delete(List<MessageInfo> infosToDelete) throws StoreException {
               StoreErrorCodes.ID_Not_Found);
-        } else if (!info.getStoreKey().isAccountContainerMatch(value.getAccountId(), value.getContainerId())) {
-          if (config.storeValidateAuthorization) {
-            throw new StoreException("DELETE authorization failure. Key: " + info.getStoreKey() + "Actually accountId: "
-                + value.getAccountId() + "Actually containerId: " + value.getContainerId(),
-                StoreErrorCodes.Authorization_Failure);
-          } else {
-            logger.warn("DELETE authorization failure. Key: {} Actually accountId: {} Actually containerId: {}",
-                info.getStoreKey(), value.getAccountId(), value.getContainerId());
-            metrics.deleteAuthorizationFailureCount.inc();
+        }
+        if (info.getLifeVersion() == MessageInfo.LIFE_VERSION_FROM_FRONTEND) {
+          // This is a delete request from frontend
+          if (!info.getStoreKey().isAccountContainerMatch(value.getAccountId(), value.getContainerId())) {
+            if (config.storeValidateAuthorization) {
+              throw new StoreException(
+                  "DELETE authorization failure. Key: " + info.getStoreKey() + "Actually accountId: "
+                      + value.getAccountId() + "Actually containerId: " + value.getContainerId(),
+                  StoreErrorCodes.Authorization_Failure);
+            } else {
+              logger.warn("DELETE authorization failure. Key: {} Actually accountId: {} Actually containerId: {}",
+                  info.getStoreKey(), value.getAccountId(), value.getContainerId());
+              metrics.deleteAuthorizationFailureCount.inc();
+            }
+          } else if (value.isDelete()) {
+            throw new StoreException(
+                "Cannot delete id " + info.getStoreKey() + " since it is already deleted in the index.",
+                StoreErrorCodes.ID_Deleted);
           }
-        } else if (value.isDelete()) {
-          throw new StoreException(
-              "Cannot delete id " + info.getStoreKey() + " since it is already deleted in the index.",
-              StoreErrorCodes.ID_Deleted);
+          indexValuesToDelete.add(value);
+          lifeVersions.add(value.getLifeVersion());
+        } else {
+          // This is a delete request from replication
+          if ((value.isDelete() && value.getLifeVersion() >= info.getLifeVersion()) || (value.getLifeVersion()
+              > info.getLifeVersion())) {
+            throw new StoreException(
+                "Cannot delete id " + info.getStoreKey() + " since it is already deleted in the index.",
+                StoreErrorCodes.Life_Version_Conflict);
+          }
+          indexValuesToDelete.add(value);
+          lifeVersions.add(info.getLifeVersion());
         }
-        indexValuesToDelete.add(value);
-        lifeVersions.add(value.getLifeVersion());
       }
       synchronized (storeWriteLock) {
         Offset currentIndexEndOffset = index.getCurrentEndOffset();
```

Comment:
It could potentially be a edge case here, when the frontend deletes a blob and replication for this delete arrive at the same time. I will check the lifeVersion before throwing an exception.